pallets/file-system/src/utils.rs

```diff
@@ -436,6 +406,37 @@ where
         Ok(msp_id)
     }
 
+    pub(crate) fn do_bsp_add_data_server_for_move_bucket_request(
+        sender: T::AccountId,
+        bucket_id: BucketIdFor<T>,
+    ) -> Result<ProviderIdFor<T>, DispatchError> {
+        let bsp_id = <T::Providers as shp_traits::ReadProvidersInterface>::get_provider_id(sender)
+            .ok_or(Error::<T>::NotABsp)?;
+
+        // Check if the sender is a Storage Provider.
+        ensure!(
+            <T::Providers as ReadStorageProvidersInterface>::is_bsp(&bsp_id),
+            Error::<T>::NotABsp
+        );
+
+        // Check if the move bucket request exists.
+        ensure!(
+            <PendingBucketsToMove<T>>::contains_key(&bucket_id),
+            Error::<T>::MoveBucketRequestNotFound,
+        );
+
+        // Check if the BSP is already a data server for the move bucket request.
+        ensure!(
+            !DataServersForMoveBucket::<T>::contains_key(&bucket_id, &bsp_id),
+            Error::<T>::BspAlreadyDataServer
+        );
+
+        // Add the data server to the move bucket request.
+        DataServersForMoveBucket::<T>::insert(&bucket_id, &bsp_id, ());
+
+        Ok(bsp_id)
+    }
```

Comment:
_pr creator_

I will remove this if there are no objections due to our recent [clearing](https://github.com/Moonsong-Labs/storage-hub/pull/241) of the notion of on chain data servers. Providers will rely solely on off-chain indexers.

Reply:
Removed